app/src/main/java/org/simple/clinic/sync/SyncDataOnApproval.kt

```diff
@@ -11,9 +13,12 @@ class SyncDataOnApproval @Inject constructor(
 
   override fun sync() {
     val shouldSync = userSession
-        .canSyncData()
-        .distinctUntilChanged()
-        .filter { canSyncData -> canSyncData }
+        .loggedInUser()
+        .filterAndUnwrapJust()
+        .map { it.status }
+        .buffer(2, 1)
```

Comment:
Why `buffer(2, 1)`

Reply:
Because we specifically want a sliding window of the last two changes to the `loggedInStatus`.

Reply:
To make it more clear, we want to do this only at the specific moment where the user's logged in status changes from anything to `ApprovedForSyncing`. However, since this is observing the database for changes, whenever the user model gets saved, this will emit.

So by buffering 2 emissions and skipping 1 value instead of 2, we get a sliding window of size 2 of the changes in the logged in status and we can explicitly check for the case where a user becomes `ApprovedForSyncing` from some other state.

Reply:
Added a comment explaining this.